src/main/src/main/java/org/geoserver/config/GeoServerLoader.java

```diff
@@ -1096,9 +1096,7 @@ void persist(XStreamPersister xp, Object obj, Resource f) throws Exception {
 
     /** Helper method which uses xstream to depersist an object as xml from disk. */
     <T> T depersist(XStreamPersister xp, Resource f, Class<T> clazz) throws IOException {
-        try (InputStream in = new ByteArrayInputStream(f.getContents())) {
-            return xp.load(in, clazz);
-        }
+        return xp.load(f, clazz);
```

Comment:
I think the BufferedInputStream does the same job and is even more efficient for large files.